internal/auth/yaml_storage.go

```diff
@@ -8,18 +8,30 @@ import (
 	"io/fs"
 	"os"
 
+	"github.com/charmbracelet/log"
 	"github.com/inovex/CalendarSync/internal/config"
 	"gopkg.in/yaml.v3"
 )
 
 type YamlStorage struct {
 	StoragePath          string
 	StorageEncryptionKey string
+	// Holds the decrypted CalendarAuth Config in memory, so the file does not have to be read multiple times
+	DecryptedAuth []CalendarAuth
 }
 
 func (y *YamlStorage) Setup(config config.AuthStorage, encryptionPassphrase string) error {
 	y.StorageEncryptionKey = encryptionPassphrase
 	y.StoragePath = config.Config["path"].(string)
+
+	log.Debug("Loading saved auth data into memory")
+	stor, err := y.readAndParseFile()
```

Comment:
What about changing this to lazy loading? Then `ReadCalendarAuth()` would store the decrypted data after loading it. That would also fix the error handling if err is not an `os.ErrNotExist`

Reply:
i had it like this, but due to my mechanism of filtering in the "ReadCalendarAuth" func the file was loaded twice. (as i returned just the auth data for the specific calendar id)

fixed.